agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/exporter/Exporter.java

```diff
@@ -276,6 +281,37 @@ private void exportRemoteDependency(SpanData span, boolean inProc) {
     exportEvents(span, null, samplingPercentage);
   }
 
+  private static final Set<String> DEFAULT_HTTP_SPAN_NAMES =
+      new HashSet<>(
+          Arrays.asList(
+              "HTTP OPTIONS",
+              "HTTP GET",
+              "HTTP HEAD",
+              "HTTP POST",
+              "HTTP PUT",
+              "HTTP DELETE",
+              "HTTP TRACE",
+              "HTTP CONNECT",
+              "HTTP PATCH"));
+
+  // the backend product prefers more detailed (but possibly infinite cardinality) name for http
+  // dependencies
+  private static String getDependencyName(SpanData span) {
+    String name = span.getName();
+
+    if (!DEFAULT_HTTP_SPAN_NAMES.contains(name)) {
+      return name;
+    }
+
+    String url = span.getAttributes().get(SemanticAttributes.HTTP_URL);
+    if (url == null) {
+      return name;
+    }
+
+    String path = UrlParser.getPathFromUrl(url);
```

Comment:
does getPathFromUrl take care of empty url? when it's empty, we want to return name as well.

Reply:
getPathFromUrl returns null currently. it has a todo.

Reply:
ya, it's just a placeholder for future work. current no-op implementation preserves existing behavior